mlrun/api/api/endpoints/migrations.py

```diff
@@ -0,0 +1,35 @@
+import http
+
+import fastapi
+
+import mlrun.api.crud
+import mlrun.api.initial_data
+import mlrun.api.schemas
+import mlrun.api.utils.background_tasks
+from mlrun.utils import logger
+
+router = fastapi.APIRouter()
+
+
+@router.post(
+    "/migrations/start",
+    responses={
+        http.HTTPStatus.OK.value: {},
+        http.HTTPStatus.ACCEPTED.value: {"model": mlrun.api.schemas.BackgroundTask},
+    },
+)
+def start_migration(
+    background_tasks: fastapi.BackgroundTasks, response: fastapi.Response,
+):
+    if mlrun.mlconf.httpdb.state == mlrun.api.schemas.APIStates.migration_in_progress:
+        raise mlrun.errors.MLRunConflictError("Migration already in progress")
+    if mlrun.mlconf.httpdb.state != mlrun.api.schemas.APIStates.waiting_for_migrations:
+        return fastapi.Response(status_code=http.HTTPStatus.OK.value)
```

Comment:
As you said we want this API to be idempotent, changed the case of `migration_in_progress` to just return the running background task.
We want that external tools, like provctl, won't need to hassle with let's check if a migration is needed, if it does, let's trigger it, but just trigger it (and if it's not needed nothing will be done)
Regarding offline - note the code in `deps.py` - offline state is the first one to be handled, it will never get here, it will return raise an exception back there already